src/commonControls/MultiValueListControl/MultiValueListControl.ts

```diff
@@ -0,0 +1,1382 @@
+     */
+    slotType: string;
+
+    /**
+     * Function(s) that determine if the value is valid.
+     *
+     * Default: `true`, i.e. any value is valid.
+     *
+     * Usage:
+     * - Validation functions return either `true` or a `ValidationResult` to
+     *   describe what validation failed.
+     */
+    validation?: SlotValidationFunction | SlotValidationFunction[];
+
+    /**
+     * List of slot-value IDs that will be presented to the user as a list.
+     */
+    listItemIDs: string[] | ((input: ControlInput) => string[]);
+
+    /**
+     * The maximum number of items spoken per turn.
+     */
+    pageSize?: number;
+
+    /**
+     * Determines if the Control must obtain a value.
+     *
+     * - If `true` the Control will take initiative to elicit a value.
+     * - If `false` the Control will not take initiative to elicit a value, but the user
+     *   can provide one if they wish, e.g. "U: My favorite color is blue".
+     */
+    required?: boolean | ((input: ControlInput) => boolean);
+
+    /**
+     * Whether the Control has to obtain explicit confirmation of the value.
+     *
+     * If `true`:
+     *  - the Control will take initiative to explicitly confirm the value with a yes/no
+     *    question.
+     */
+    confirmationRequired?: boolean | ((input: ControlInput) => boolean);
+
+    /**
+     * Props to customize the prompt fragments that will be added by
+     * `this.renderAct()`.
+     */
+    prompts?: MultiValueListControlPromptProps;
+
+    /**
+     * Props to customize the reprompt fragments that will be added by
+     * `this.renderAct()`.
+     */
+    reprompts?: MultiValueListControlPromptProps;
+
+    /**
+     * Props to customize the relationship between the control and the
+     * interaction model.
+     */
+    interactionModel?: MultiValueListControlInteractionModelProps;
+
+    /**
+     * Props to configure input handling.
+     */
+    inputHandling?: ControlInputHandlingProps;
+
+    /**
+     * Function that maps the MultiValueListControlState.value to rendered value that
+     * will be presented to the user as a list.
+     *
+     * Default: returns the value unchanged.
+     */
+    valueRenderer?: (value: string[], input: ControlInput) => string[];
+
+    /**
+     * Props to customize the APL generated by this control.
+     */
+    apl?: MultiValueListControlAPLProps;
+}
+
+/**
+ * ListControl validation function
+ */
+export type SlotValidationFunction = (
+    values: MultiValueListStateValue[],
+    input: ControlInput,
+) => true | MultiValueValidationFailure | Promise<true | MultiValueValidationFailure>;
+
+export type AplContent = { document: { [key: string]: any }; dataSource: { [key: string]: any } };
+export type AplContentFunc = (control: MultiValueListControl, input: ControlInput) => AplContent;
+export type AplDocumentPropNewStyle = AplContent | AplContentFunc;
+
+/**
+ * Mapping of action slot values to the behaviors that this control supports.
```

Comment:
wording: behavior -> capability.

(pls also update the same for other controls as the current wording is confusing)